src/component/Text.js

```diff
@@ -0,0 +1,162 @@
+import React, { Component, PropTypes } from 'react';
+import reduceCSSCalc from 'reduce-css-calc';
+import { PRESENTATION_ATTRIBUTES } from '../util/ReactUtils';
+import _ from 'lodash';
+
+class Text extends Component {
+
+  static propTypes = {
+    ...PRESENTATION_ATTRIBUTES,
+    fit: PropTypes.bool,
+    angle: PropTypes.number,
+    textAnchor: PropTypes.oneOf(['start', 'middle', 'end', 'inherit']),
+    verticalAnchor: PropTypes.oneOf(['start', 'middle', 'end']),
+  };
+
+  static defaultProps = {
+    x: 0,
+    y: 0,
+    lineHeight: '1em',
+    capHeight: '0.71em', // Magic number from d3
+    fit: false,
+    textAnchor: 'start',
+    verticalAnchor: 'end', // Maintain compat with existing charts / default SVG behavior
+  };
+
+  state = {
+    wordsByLines: [],
+  };
+
+  componentWillMount() {
+    this.updateWordsByLines(this.props, true);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const calculateWordWidths = (
+      this.props.children !== nextProps.children ||
+      this.props.style !== nextProps.style
+    );
+    this.updateWordsByLines(nextProps, calculateWordWidths);
+  }
+
+  updateWordsByLines(props, calculateWordWidths) {
+    // Only perform calculations if using features that require them (multiline, fit)
+    if (props.width || props.fit) {
+      if (calculateWordWidths) {
+        const { wordsWithComputedWidth, spaceWidth } = this.calculateWordWidths(props);
+        this.wordsWithComputedWidth = wordsWithComputedWidth;
+        this.spaceWidth = spaceWidth;
+      }
+
+      const wordsByLines = this.calculateWordsByLines(
+        this.wordsWithComputedWidth,
+        this.spaceWidth,
+        props.width
+      );
+      this.setState({ wordsByLines });
+    } else {
+      const words = props.children ? props.children.toString().split(/\s+/) : [];
+      this.setState({ wordsByLines: [{ words }] });
+    }
+  }
+
+  calculateWordWidths(props) {
+    // Calculate length of each word to be used to determine number of words per line
+    var svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+    var text = document.createElementNS('http://www.w3.org/2000/svg', 'text');
+    _.assign(text.style, props.style);
```

Comment:
I tried this originally but it appears `text.style` does not allow assignment, just it's properties.  I also had `Object.assign` at one point but found out it wasn't polyfilled and thus failed on some older browsers.